### PR TITLE
FAD-5897 - Templates Fixes

### PR DESCRIPTION
**Pull request by jonambas**

### CODE REVIEW

File: src/pages/templates/components/FromEmail.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      downshift.setHighlightedIndex(0);
+    }
+  }
+
+  getMatches = (inputValue, selectedItem) => {
+    const parts = inputValue.split('@');
+    let matches = [];
+
+    // Show matches as soon as '@' exists
+    if (parts[1] === '' || parts[1]) {
```
Comment (jonambas) [inline src/pages/templates/components/FromEmail.js]: Can maybe do `parts.length > 1`

File: src/pages/templates/components/FromEmail.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    const { onChange } = this.props;
+
+    // Push changes to redux form store
+    if (changes.hasOwnProperty('inputValue')) {
+      onChange && onChange(changes.inputValue);
+    }
+
+    // Highlights first item in list by default
+    if (!downshift.highlightedIndex) {
+      downshift.setHighlightedIndex(0);
```
Comment (jonambas) [inline src/pages/templates/components/FromEmail.js]: #165 Should do this on all other typeaheads too

File: src/pages/templates/components/FromEmail.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      index, item,
+      content: item,
+      highlighted: highlightedIndex === index
+    }));
+
+    const textFieldProps = getInputProps({ ...rest, value: this.state.value });
+    const listClasses = cx(styles.List, (isOpen && matches.length) && styles.open);
+
+    return (
+      <div className={styles.Typeahead} onFocus={this.handleFocus}>
```
Comment (jonambas) [inline src/pages/templates/components/FromEmail.js]: remove `this.handleFocus`

File: src/pages/templates/components/Form.js [hunk 4]
```diff
@@ -69,10 +96,11 @@ class Form extends Component {
 
           <Field
             name='content.from.email'
-            component={TextFieldWrapper}
+            component={FromEmailWrapper}
             label='From Email'
-            disabled={newTemplate || published} // TODO check for sending domains
-            validate={[required, emailOrSubstitution, verifiedDomain]}
+            disabled={!domains.length || published}
+            validate={[required, emailOrSubstitution, this.verifiedDomain]}
```
Comment (rajumsys) [inline src/pages/templates/components/Form.js]: seems, it's not complaining about invalid from email like `from*&@DOMAIN`

**Comment by rajumsys** [comment]
A few initial feedbacks:
- Initial impression was little confusing. I was looking where is domain selector/drop down. But it makes sense after typing `@`. Can a placeholder be useful to avoid confusing before that?
- Seems I need to click twice on a drop down item to actually be selected. On first click, it shows validation error
> Must use a verified sending domain

**Comment by jonambas** [comment]
@rajumsys Seems like the invalid email is coming from bad regex in `emailRegex` in `helpers/regex.js`. I'm very bad with regex, so I'm not sure how to fix this:
```
export const emailRegex = /^[a-z0-9!#$%&'*+/=?^_`{|}~.-]+@[a-z0-9]([a-z0-9-]*[a-z0-9])?(\.[a-z0-9]([a-z0-9-]*[a-z0-9])?)*$/i;
```

**Pull request merged**